### PR TITLE
Add the function subjectIDsValid to profileTemplate

### PR DESCRIPTION
**Pull request by cmadjar**
### Description

Currently, the `getSubjectIDs` function of `MRI.pm` is calling `sujectIDsValid` that checks:
- if the PSCID/CandID refers to a valid candidate 
- if the Visit Label is present in the `Visit_Windows` (important to check when using the Create Visit label flag)

### This fixes

- [x] Github #410 

### Caveat for existing projects

If the function `getSubjectIDs` of the MRI configuration file (typically called `prod` in `dicom-archive/.loris-mri/`) has the following flag `$subjectID{'createVisitLabel'}` set to `1` (`$subjectID{'createVisitLabel'} = 1;`), then you might want to modify the function to include the code changes made in `profileTemplate` of this PR.

A.K.A.: modify
```
     # If patient match PSCID_DCCID_VisitLabel
     # Determine PSCID, DCCID and visitLabel based on patient name
    } elsif ($patientName =~ /([^_]+)_(\d+)_([^_]+)/) {

        $subjectID{'PSCID'}         = NeuroDB::MRI::my_trim($1);
        $subjectID{'CandID'}        = NeuroDB::MRI::my_trim($2);
        $subjectID{'visitLabel'}    = NeuroDB::MRI::my_trim($3);
        $subjectID{'createVisitLabel'} = 1;

        print "PSCID is: "            . $subjectID{'PSCID'}      .
                "\n CandID id: "      . $subjectID{'CandID'}     .
                "\n visit_label is: " . $subjectID{'visitLabel'} . "\n";

    }
```
to 
```
     # If patient match PSCID_DCCID_VisitLabel
     # Determine PSCID, DCCID and visitLabel based on patient name
    } elsif ($patientName =~ /([^_]+)_(\d+)_([^_]+)/) {

        $subjectID{'PSCID'}         = NeuroDB::MRI::my_trim($1);
        $subjectID{'CandID'}        = NeuroDB::MRI::my_trim($2);
        $subjectID{'visitLabel'}    = NeuroDB::MRI::my_trim($3);
        $subjectID{'createVisitLabel'} = 1;

        # Note, this function will make sure that the PSCID/CandID information refers to a valid candidate
        # and it will check that the Visit label exists in the Visit_Windows table
        if(!NeuroDB::MRI::subjectIDIsValid($subjectID{'CandID'}, $subjectID{'PSCID'}, $subjectID{'visitLabel'}, $dbhr)) {
            return undef;
        }

        print "PSCID is: "            . $subjectID{'PSCID'}      .
                "\n CandID id: "      . $subjectID{'CandID'}     .
                "\n visit_label is: " . $subjectID{'visitLabel'} . "\n";

    }

```

### CODE REVIEW

**Comment by cmadjar** [comment]
@jesscall I assigned this PR to you as this may fix the issue you are having with CCNA creating wrong visit labels when inserting a scan that was not properly labelled in terms of Visit label.

Let me know if you have any questions.

File: dicom-archive/profileTemplate [hunk 1]
```diff
@@ -58,6 +58,12 @@ sub getSubjectIDs {
         $subjectID{'visitLabel'}    = NeuroDB::MRI::my_trim($3);
         $subjectID{'createVisitLabel'} = 0; 
 
+        # Note, this function will make sure that the PSCID/CandID information refers to a valid candidate
```
Comment (PapillonMcGill) [inline dicom-archive/profileTemplate]:
```suggestion
        # Note, this function will ensure that the PSCID/CandID information refers to a valid candidate
```

Comment (cmadjar) [reply dicom-archive/profileTemplate]: done

**Comment by cmadjar** [comment]
@PapillonMcGill Thank you for your review! Sorry it took so long for me to integrate your comment. I added a documentation part also (for easier viewing, click on the view file to be able to see the troubleshooting table). Please re-review at your convenience.

Thanks again!

File: docs/AppendixA-Troubleshooting_guideline.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ _**Table 3: Common errors encountered during execution of the LORIS-MRi insertio
...
 |`no MINCs inserted`|Possibly all the MINC images are violated scans|Check the details of the image headers (from the MRI Violated Scans module or using `mincheader`) against the `mri_protocol` table entries, and adjust the table protocol parameters accordingly|
-
+|The pipeline created an invalid visit label in the `session` table when inserting a scan (a.k.a. Visit Label not listed in the `Visit_Windows` table)|The flag $subjectID{'createVisitLabel'} is set to 1 but the function `getSubjectIDs` of the profile file does not contain a call to validate the subject IDs information|Make sure that the `getSubjectIDs` function of your profile file (typically named `prod`) contains a call to the function `NeuroDB::MRI::subjectIDIsValid` on the `CandID`, `PSCID` and `VisitLabel` values (see https://github.com/aces/Loris-MRI/pull/411 for details)|
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/AppendixA-Troubleshooting_guideline.md]: @PapillonMcGill Done! Please, re-review. Thanks!

**Comment by cmadjar** [comment]
@PapillonMcGill WOW! That was fast :). Thanks!!

**Comment by PapillonMcGill** [comment]
Effectiveness!

**Pull request merged**